File: src/contests/cg/puzzle/vbnet/rubik.vb
' 🎮 CodinGame Puzzle - rubik
' https://www.codingame.com/training/medium/rubik

Imports System

Module Solution
    Sub Main()
        Dim n As Long = Long.Parse(Console.ReadLine().Trim())
        ' Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
        Dim inner As Long = If(n >= 2, n - 2, 0L)
        Console.WriteLine(n * n * n - inner * inner * inner)
    End Sub
End Module

File: src/contests/cg/puzzle/fsharp/rubik.fs
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

let n = int64 (System.Console.ReadLine().Trim())
// Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
let inner = if n >= 2L then n - 2L else 0L
printfn "%d" (n * n * n - inner * inner * inner)
